Algorithms-Merge-Sort.md

```diff
@@ -0,0 +1,53 @@
+# Merge Sort
+
+Most modern languages have an inbuilt sort() function which automatically sorts an input array or list. Did you ever wonder how the sort function actually works on the inside?. Knowing common sorting algorithms and their implementations is the most important part of a coding interview. In this series of articles, we will look at several important sorting algorithms. How they are implemented, the time and space complexity etc. Our very first post is on Merge Sort.
+
+To learn about Merge Sort, a basic knowledge about Recursion is a pre-requisite. Merge Sort is based on the principle of Divide and Conquer. The whole process of sorting an array of N integers can be summarized into three steps-
+
+- Divide the array into two halves.
+- Sort the left half and the right half using the same recurring algorithm.
+- Merge the sorted halves.
+
+The biggest advantage of using Merge sort is that it takes only n*log(n) steps to sort an entire Array. It is a lot better than n^2 running time of bubble sort or insertion sort.
+Before we write the JavaScript code, let us understand how merge sort works with the help of a diagram.
+
+![alt tag] (http://i67.tinypic.com/2ahe49y.png)
+
+- Initially we have an array of 6 unsorted integers Arr(5, 8, 3, 9, 1, 2)
+- We split the array into two halves Arr1 = (5, 8, 3) and  Arr2 = (9, 1, 2).
+-  Again, we divide them into two halves: Arr3 = (5, 8) and Arr4 = (3) and  Arr5 = (9, 1) and Arr6 = (2)
+-  Again, we divide them into two halves: Arr7 = (5), Arr8 =  (8),  Arr9 = (9),  Arr10 = (1) and Arr6 = (2)
+-  We will now compare the elements in these sub arrays in order to merge them. 
+
+Let's write MergeSort in JavaScript:
+
+```
+function mergeSort (arr) {    
+  if (arr.length < 2) return arr;
+  var mid = Math.floor(arr.length /2);
+  var subLeft = mergeSort(arr.slice(0,mid));
+  var subRight = mergeSort(arr.slice(mid));
+  return merge(subLeft, subRight);
+  }
+  
+```
```

Comment:
provide appropriate [code block](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#code-and-syntax-highlighting) for better syntax highlighting. Like for JavaScript its ````javascript`